lib/holidays/use_case/context/between.rb

```diff
@@ -13,102 +13,16 @@ def call(start_date, end_date, dates_driver, regions, observed, informal)
-                      valid_range_year = true if year_range[year_range.keys.first].cover?(year)
-                    end
-                    break if valid_range_year
-                  end
-                  next unless valid_range_year
-                end
-
-                #FIXME I don't like this entire if/else. If it's a function, do something, else do some
-                # weird mday logic? Bollocks. I think this should be a refactor target.
-                if h[:function]
-                  function_arguments = []
-
-                  #FIXME This is a refactor target. We should also allow 'date'. Right now these are the only
-                  # three things that we allow in. I think having a more testable, robust approach here is vital.
-                  if h[:function_arguments].include?(:year)
-                    function_arguments << year
-                  end
-
-                  if h[:function_arguments].include?(:month)
-                    function_arguments << month
-                  end
-
-                  if h[:function_arguments].include?(:day)
-                    function_arguments << h[:mday]
-                  end
-
-                  result = call_proc(h[:function], *function_arguments)
-
-                  #FIXME This is a dangerous assumption. We should raise an error or something
-                  #      if these procs return something unexpected.
-                  #
-                  # Procs may return either Date or an integer representing mday
-                  if result.kind_of?(Date)
-                    if h[:function_modifier]
-                      result = result + h[:function_modifier] # NOTE: This could be a positive OR negative number.
-                    end
-
-                    month = result.month
-                    mday = result.mday
-                  else
-                    mday = result
-                  end
-                else
-                  mday = h[:mday] || day_of_month_calculator.call(year, month, h[:week], h[:wday])
-                end
-
-                # Silently skip bad mdays
-                begin
-                  date = Date.civil(year, month, mday)
-                rescue; next; end
-
-                #FIXME We should be checking the function arguments and passing in what is specified.
-                # Right now all 'observed' functions require 'date' but that is by convention. Nothing
-                # is requiring that. We should be more explicit.
-                if observed && h[:observed]
-                  date = call_proc(h[:observed], date)
-                end
-
-                if date.between?(start_date, end_date)
-                  holidays << {:date => date, :name => h[:name], :regions => h[:regions]}
-                end
-              end
-            end
-          end
-
+          holidays = holidays.select{|holiday|holiday[:date].between?(start_date, end_date)}
```

Comment:
I have read your PR. It's great!! 👍 
I think each contest class should have low dependency , because it's easier to maintain source code.
Thank for your review. 😃